da/mock/mock.go

```diff
@@ -80,10 +85,16 @@ func (m *MockDataAvailabilityLayerClient) CheckBlockAvailability(header *types.H
 // RetrieveBlock returns block at given height from data availability layer.
 func (m *MockDataAvailabilityLayerClient) RetrieveBlock(height uint64) da.ResultRetrieveBlock {
 	hash, err := m.dalcKV.Get(getKey(height))
+	if errors.Is(err, badger.ErrKeyNotFound) {
```

Comment:
Same here.

Reply:
```suggestion
	if errors.Is(err, store.ErrKeyNotFound) {
```

Reply:
In `RetrieveBlock` there is no need to distinguish `ErrKeyNotFound`, as in this context it's error as any other.

Reply:
Great.